mmdet/structures/bbox/base_bbox.py

```diff
@@ -0,0 +1,273 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+from abc import ABCMeta, abstractclassmethod
+from inspect import signature
+from typing import Callable, Optional, Union
+
+import numpy as np
+import torch
+from torch import Tensor
+
+
+class BaseBBoxes(metaclass=ABCMeta):
+    """Base class for bounding boxes.
+
+    Args:
+        bboxes(Tensor or np.ndarray): The bbox data.
+        src_submode (str, optional): The submode of bboxes. BaseBBoxes will
+            according to src_submode to convert bboxes into regular format.
+            Defaults to None.
+        device(str or torch.device): device of bboxes.
+        dtype(torch.dtype): data type of bboxes.
+    """
+
+    _bbox_dim: int = 0
+    SUBMODE_CONVERTERS: dict = None
+
+    def __init__(self,
+                 bboxes: Union[Tensor, np.ndarray],
+                 src_submode: Optional[str] = None,
+                 device: Union[str, torch.device] = None,
+                 dtype: torch.dtype = None) -> None:
+        if isinstance(bboxes, np.ndarray):
+            bboxes = torch.from_numpy(bboxes)
```

Comment:
is that possible to add a function to covert bbox from torch to numpy, for example 
```
bbox.numpy()
```